packages/extension-core/src/domains/accounts/handler.ts

```diff
@@ -1,482 +1,208 @@
   }
 
   private async accountCreateSuri({
     name,
     suri,
-    type,
+    curve = "sr25519",
   }: RequestAccountCreateFromSuri): Promise<string> {
     const password = await this.stores.password.getPassword()
     assert(password, "Not logged in")
 
-    const expectedAddress = addressFromSuri(suri, type)
-
-    const notExists = !keyring
-      .getAccounts()
-      .some((acc) => acc.address.toLowerCase() === expectedAddress.toLowerCase())
-    assert(notExists, "Account already exists")
-
-    //suri includes the derivation path if any
-    const splitIdx = suri.indexOf("/")
-    const mnemonic = splitIdx === -1 ? suri : suri.slice(0, splitIdx)
-    const derivationPath = splitIdx === -1 ? "" : suri.slice(splitIdx)
-
-    const meta: KeyringPair$Meta = {
-      name,
-      origin: AccountType.Talisman,
-    }
-
-    // suri could be a private key instead of a mnemonic
-    if (mnemonicValidate(mnemonic)) {
-      const derivedMnemonicId = await this.stores.mnemonics.getExistingId(mnemonic)
-
-      if (derivedMnemonicId) {
-        meta.derivedMnemonicId = derivedMnemonicId
-        meta.derivationPath = derivationPath
-      } else {
-        const result = await this.stores.mnemonics.add(
-          `${name} Recovery Phrase`,
-          mnemonic,
-          password,
-          MnemonicSource.Imported,
-          true,
-        )
-        if (result.ok) {
-          meta.derivedMnemonicId = result.val
-          meta.derivationPath = derivationPath
-        } else throw new Error("Failed to store mnemonic", { cause: result.val })
-      }
-    } else {
-      meta.importSource = AccountImportSources.PK
+    // throws if invalid mnemonic
+    const parsedSuri = parseSuri(suri)
+    if (parsedSuri.password) {
+      // TODO: to support this properly, dont store mnemonic and just create it as a keypair that doesnt have a mnemonic
+      throw new Error("Password not supported for suri")
     }
 
-    try {
-      const { pair } = keyring.addUri(
-        suri,
-        password,
-        meta,
-        type, // if undefined, defaults to keyring's default (sr25519 atm)
-      )
-
-      this.captureAccountCreateEvent(type, "seed")
-
-      return pair.address
-    } catch (error) {
-      throw new Error((error as Error).message)
-    }
-  }
+    //suri includes the derivation path if any
+    const { mnemonic, derivationPath } = parseSuri(suri)
 
-  private async accountCreateJson({
-    unlockedPairs,
-  }: RequestAccountCreateFromJson): Promise<string[]> {
-    const password = await this.stores.password.getPassword()
-    assert(password, "Not logged in")
+    let mnemonicId = await keyringStore.getExistingMnemonicId(mnemonic)
 
-    const addresses: string[] = []
-    for (const json of unlockedPairs) {
-      const pair = keyring.createFromJson(json, {
-        name: json.meta?.name || "Json Import",
-        origin: AccountType.Talisman,
-        importSource: AccountImportSources.JSON,
+    if (!mnemonicId) {
```

Comment:
I think we can reuse the `parsedSuri` var?

```suggestion

    // suri includes the derivation path if any
    const { mnemonic, derivationPath } = parsedSuri

    let mnemonicId = await keyringStore.getExistingMnemonicId(mnemonic)
    if (!mnemonicId) {
```

Reply:
Diff is

```diff
-    //suri includes the derivation path if any
-    const { mnemonic, derivationPath } = parseSuri(suri)
+    // suri includes the derivation path if any
+    const { mnemonic, derivationPath } = parsedSuri
```